Source/DafnyLanguageServer/Workspace/DafnyDocument.cs

```diff
@@ -32,6 +34,35 @@ public record DafnyDocument(
     /// </summary>
     public string? SerializedCounterExamples { get; init; }
 
+
+    /// <summary>
+    /// True is the resolution succeeded, false if resolution failed
+    /// <c>null</c> If the verification did not start (e.g. because of resolution errors)
+    /// </summary>
+    public bool? ResolutionSucceeded { get; set; } = null;
+
+    /// <summary>
+    /// True is the verification pass went through, false if it failed because of verification errors
+    /// <c>null</c> If the verification did not start (e.g. because of resolution errors or verification deactivated)
+    /// </summary>
+    public bool? VerificationPass { get; set; } = null;
```

Comment:
In the case of on-save verification, I know that verification is deactivated, and I don't trigger the gutter highlight on the client.